google/resource_composer_environment_test.go

```diff
@@ -0,0 +1,506 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+
+	if env.Config == nil {
+		t.Fatalf("expected read value to have non-nil config")
+	}
+
+	if env.Config.SoftwareConfig == nil {
+		t.Fatalf("expected non-nil SoftwareConfig")
+	}
+
+	if len(env.Config.SoftwareConfig.PypiPackages) != 1 {
+		t.Errorf(`expected PypiPackages to have one key-value { "scipy": "==1.1.0" }, got: %#v`, env.Config.SoftwareConfig.PypiPackages)
+	} else if v, ok := env.Config.SoftwareConfig.PypiPackages["scipy"]; !ok || v != "==1.1.0" {
+		t.Errorf(`expected PypiPackages to contain { "scipy": "==1.1.0" }, got: %#v`, env.Config.SoftwareConfig.PypiPackages)
+	}
+}
+
+func testAccCheckComposerEnvironmentExists(n string, environment *composer.Environment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		idTokens := strings.Split(rs.Primary.ID, "/")
+		if len(idTokens) != 3 {
+			return fmt.Errorf("Invalid ID %q, expected format {project}/{region}/{environment}", rs.Primary.ID)
+		}
+		envName := &composerEnvironmentName{
+			Project:     idTokens[0],
+			Region:      idTokens[1],
+			Environment: idTokens[2],
+		}
+
+		nameFromId := envName.resourceName()
+		config := testAccProvider.Meta().(*Config)
+
+		found, err := config.clientComposer.Projects.Locations.Environments.Get(nameFromId).Do()
+		if err != nil {
+			return err
+		}
+
+		if found.Name != nameFromId {
+			return fmt.Errorf("Environment not found")
+		}
+
+		*environment = *found
+		return nil
+	}
+}
+
+func testAccComposerEnvironmentDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_composer_environment" {
+			continue
+		}
+
+		idTokens := strings.Split(rs.Primary.ID, "/")
+		if len(idTokens) != 3 {
+			return fmt.Errorf("Invalid ID %q, expected format {project}/{region}/{environment}", rs.Primary.ID)
+		}
+		envName := &composerEnvironmentName{
+			Project:     idTokens[0],
+			Region:      idTokens[1],
+			Environment: idTokens[2],
+		}
+
+		_, err := config.clientComposer.Projects.Locations.Environments.Get(envName.resourceName()).Do()
+		if err == nil {
+			return fmt.Errorf("environment %s still exists", envName.resourceName())
+		}
+	}
+
+	return nil
+}
+
+func testAccComposerEnvironment_basic(name string) string {
+	return fmt.Sprintf(`
+resource "google_composer_environment" "test" {
+  name           = "%s"
+  region         = "us-central1"
+}
+`, name)
+}
+
+func testAccComposerEnvironment_update(name string) string {
+	return fmt.Sprintf(`
+resource "google_composer_environment" "test" {
+	name = "%s"
+	region = "us-central1"
+
+	config {
+		node_count = 4
+
+		software_config {
+			airflow_config_overrides {
+			  core-load_example = "True"
```

Comment:
nothing to change here but wow nice mix of hyphens and underscores, airflow.

Reply:
yeah (also commenting for last-week-i-learned), the way gcp decided to do the naming was ${section name}-${actual_snake_case_property} - the actual airflow config then looks like 
```
[core]
load_example = True
```